pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -107,16 +131,38 @@ func initDrainer(r *ReconcileNodeMaintenance, config *rest.Config) error {
 	//Label selector to filter pods on the node
 	//r.drainer.PodSelector = "kubevirt.io=virt-launcher"
 
-	cs, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return err
-	}
+
 	r.drainer.Client = cs
 	r.drainer.DryRun = false
 
+	isLeaseSupported, err := checkLeaseSupported(r.drainer.Client);
+	if err != nil {
+		log.Errorf("Error while checking for lease support %v", err)
+	}
+
+	r.isLeaseSupported = isLeaseSupported
+
 	return nil
 }
 
+func checkLeaseSupported(cs kubernetes.Interface) (bool, error) {
+
+	groupList, err := cs.Discovery().ServerGroups()
+	if err != nil {
+		return false, err
+	}
+	isLeaseSupported := false
+	if groupList != nil {
+		apiVersions := metav1.ExtractGroupVersions(groupList)
+		for _, v := range apiVersions {
+			if v == "coordination.k8s.io/v1beta1" {
```

Comment:
please make strings such  "coordination.k8s.io/v1beta1"  as constants. 
what happens if version is changes here ?  what happens if we upgrade to a newer version?

Reply:
done